drivers/sensor/ina700/ina700.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2024 Esco Medical ApS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_ina700
+
+#include "ina700.h"
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(INA700, CONFIG_SENSOR_LOG_LEVEL);
+
+static void ina700_lock_data(const struct device *dev);
+static void ina700_unlock_data(const struct device *dev);
+
+static int reg_read_24(const struct device *dev, uint8_t addr, uint32_t *data)
+{
+	const struct ina700_config *cfg = dev->config;
+	uint8_t rx_buf[3];
+
+	int rc = i2c_burst_read_dt(&cfg->bus, addr, rx_buf, sizeof(rx_buf));
+	*data = sys_get_be24(rx_buf);
+
+	return rc;
+}
+
+static int reg_read_16(const struct device *dev, uint8_t addr, uint16_t *data)
+{
+	const struct ina700_config *cfg = dev->config;
+	uint8_t rx_buf[2];
+
+	int rc = i2c_burst_read_dt(&cfg->bus, addr, rx_buf, sizeof(rx_buf));
+	*data = sys_get_be16(rx_buf);
+
+	return rc;
+}
+
+static int reg_write_16(const struct device *dev, uint8_t addr, uint16_t data)
+{
+	const struct ina700_config *cfg = dev->config;
+	uint8_t tx_buf[3];
+
+	tx_buf[0] = addr;
+	sys_put_be16(data, &tx_buf[1]);
+
+	return i2c_write_dt(&cfg->bus, tx_buf, sizeof(tx_buf));
+}
+
+static int ina700_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	int rc = 0;
+	struct ina700_data *data = dev->data;
+
+	ina700_lock_data(dev);
+
+	if ((chan == SENSOR_CHAN_VOLTAGE) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_16(dev, INA700_REG_VBUS, &data->voltage);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+	if ((chan == SENSOR_CHAN_CURRENT) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_16(dev, INA700_REG_CURRENT, &data->current);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+	if ((chan == SENSOR_CHAN_POWER) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_24(dev, INA700_REG_POWER, &data->power);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+	if ((chan == SENSOR_CHAN_DIE_TEMP) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_16(dev, INA700_REG_DIE_TEMPERATURE, &data->temperature);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+exit:
+	ina700_unlock_data(dev);
+	return 0;
+}
+
+static int ina700_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	int rc = 0;
```

Comment:
Not assigned before being read :)
```suggestion
	int rc;
```